angular/src/providers/nav-controller.ts

```diff
@@ -169,6 +169,7 @@ export class NavController {
    * @internal
    */
   consumeTransition() {
+    let isDirectionDefined = this.direction === 'forward' || this.direction === 'back';
```

Comment:
This check can probably be done in the `setActive` function in `stack-controller.ts` so we dont need to export more data from this function. We should be able to do `direction !== 'root'`.